src/ngraph/ngraph_graph.h

```diff
@@ -202,6 +202,7 @@ class Graph : public Node {
         context_(context),
         enable_fprop_cache(enable_fprop_cache) {
     fprop_cache = std::make_shared<ngraph::FpropCache>();
+    is_reuse_mem = context.dev_type != mxnet::Context::kCPU;
```

Comment:
Can't GPU utilize mem reuse?

Reply:
Actually, as written, "Can nothing but CPU utilize memory reuse?"

Reply:
we compile graph targeted at specific backend. Since this flag is graph level, we can choose if mem_reuse should apply or not. At this time, we only anticipate one backend that cant reuse mem : NNP. All other backends can reuse mxnet mem (eg: cpu, gpu etc). 

The above line will be changed to kNNP, it is currently set to kCPU just for CI testing.

Reply:
@diyessi agree, we were expecting ngraph to handle this transparently. 
This PR is mxnet temporary fix, until ngraph handles NNP mem-reuse.

Reply:
Please add a JIRA so it gets addressed. JIRA remembers things better than I do.